DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -43,59 +51,59 @@ internal class RecordingCoordinator {
         recorder: Recording,
         sampler: Sampler,
         telemetry: Telemetry,
-        startRecordingImmediately: Bool,
-        methodCallTelemetrySamplingRate: Float = 0.1
-    ) {
+        recordingTrigger: RecordingTriggering,
+        methodCallTelemetrySamplingRate: Float = 0.1,
+        dateProvider: DateProvider = SystemDateProvider(),
+        queue: Queue = MainAsyncQueue()
+    ) throws {
```

Comment:
Would it make sense to add some telemetry in cases the swizzling or recorder initialization fails? This could help in diagnosing potential customer issues.